cpp/tests/streams/stream_compaction_test.cpp

```diff
@@ -377,6 +378,20 @@ TEST_F(StreamCompactionTest, ApplyBooleanMask)
   CUDF_TEST_EXPECT_TABLES_EQUAL(expected, *result);
 }
 
+TEST_F(StreamCompactionTest, ASTFilterJitExec)
```

Comment:
I think this should go with the filter tests and not in `stream_compaction`.
You may want to create a separate `filter_test.cpp` in the `streams` directory for these.

Reply:
`filter` just like `apply_boolean_mask` is in the `stream_compaction` header. do you think it's worth separating?

Reply:
Ah yes, you are right. I missed that filter was in the stream compaction header.